src/app/components/pickers/dateRange.tsx

```diff
@@ -34,6 +34,11 @@ const DateRange = ({
   const [startDate, setStartDate] = useState(initialStartDateValue);
   const [endDate, setEndDate] = useState(initialEndDateValue);
 
+  useEffect(() => {
+    setStartDate(initialStartDateValue);
+    setEndDate(initialEndDateValue);
+  }, [initialStartDateValue, initialEndDateValue]);
+
```

Comment:
I think this `useEffect` is not required, as prop changes the state also updated